data/templates/set_bmc_credentials.sh

```diff
@@ -47,14 +49,38 @@ for x in $cmdReturn; do
   i=$((i+1))
 done
 
+function get_newUserNumber()
+{
+        cmdReturn=$(sudo ipmitool user summary $channel)
+        myarray=(${cmdReturn//$'\n'/ })
+        MaxCount=${myarray[3]}
+        if [ $userNumber -lt $MaxCount ]; then
+                #try to find out the empty user id
+                maxLength=${#userlist[@]}
+                for ((i=1;i<$maxLength;i++)) do
+                        ID=`echo ${userlist[i]} | awk '{print $1}'`
+                        if [ $ID != $i ]; then
+                                newUserNumber=$i
+                                break
+                        fi
+                done
+                if [ $newUserNumber -eq 0 ]; then
+                        newUserNumber=$((userNumber + 1))
+                fi
+        else
+                ecoh 'reach max user count'
+                exit 1
+        fi
+}
 
 if [ $check == 0 ]; then
  echo "Creating a new user"
- userNumber=$((userNumber + 1))
- ipmitool user set name $userNumber <%=user%>
- ipmitool user set password $userNumber <%=password%>
- ipmitool channel setaccess $channel $userNumber callin=on ipmi=on link=on privilege=4
- ipmitool user enable $userNumber
+ get_newUserNumber
+# userNumber=$((userNumber + 1))
```

Comment:
I think what @yyscamper means is that you need to remove the line 79
`# userNumber=$((userNumber + 1))`